client/src/components/CrossCheck/CrossCheckCriteriaForm.tsx

```diff
@@ -40,44 +40,44 @@ export interface CriteriaFormProps {
   countStar: CountState[];
   setCountStar: (newCount: CountState[]) => void;
   criteriaData: CrossCheckCriteriaData[];
-  totalPoints: number;
-  setTotalPoints: (newPoint: number) => void;
+  totalPoints?: number;
   penalty: CountState[];
   setPenalty: (newPenalty: CountState[]) => void;
   criteriaComment: CommentState[];
   setComment: (newComment: CommentState[]) => void;
-  maxScoreForTask: number;
 }
 
 export function CrossCheckCriteriaForm({
   countStar,
   setCountStar,
   criteriaData,
   totalPoints,
-  setTotalPoints,
   penalty,
   setPenalty,
   criteriaComment,
   setComment,
-  maxScoreForTask,
 }: CriteriaFormProps) {
+  const form = Form.useFormInstance();
   const penaltyData: CrossCheckCriteriaData[] =
     criteriaData?.filter(item => item.type.toLowerCase() === TaskType.Penalty).map(item => omit(item, ['point'])) ?? [];
 
-  function calculateTotalPoints() {
-    return useMemo(() => {
-      const sumPoints = +countStar
-        .map(item => item.point)
-        .reduce((prev, next) => prev + next, 0)
-        .toFixed(1);
-      const sumPenalty = penalty.map(item => item.point).reduce((prev, next) => prev + next, 0);
-      const finalPoints = sumPoints + sumPenalty;
-      setTotalPoints(finalPoints > 0 ? finalPoints : 0);
-      return finalPoints > 0 ? finalPoints : 0;
-    }, [countStar, penalty]);
-  }
+  useEffect(() => {
+    form.setFieldValue('score', totalPoints);
+  }, [totalPoints]);
+
+  useEffect(() => {
+    if (criteriaData.length === 0) return;
 
-  const calculationResultPoints = calculateTotalPoints();
+    const sumPoints = +countStar
+      .map(item => item.point)
+      .reduce((prev, next) => prev + next, 0)
+      .toFixed(1);
+    const sumPenalty = penalty.map(item => item.point).reduce((prev, next) => prev + next, 0);
+    let finalPoints = sumPoints + sumPenalty;
+    finalPoints = finalPoints > 0 ? finalPoints : 0;
+
+    form.setFieldValue('score', finalPoints);
```

Comment:
Not 100% related to your changes, but still:
1. We could remove excessive `map` iteration, `reduce` is enough here
2. We could avoid usage of `let` here, it's not necessary :)
I would change it to something like this
```suggestion
    const scorePoints = +countStar
      .reduce((acc, next) => acc + next.point, 0)
      .toFixed(1);

    const penaltyPoints = penalty.reduce((acc, next) => acc + next.point, 0);

    const totalScore = scorePoints + penaltyPoints;
    const finalScore = totalScore > 0 ? totalScore : 0;

    form.setFieldValue('score', finalScore);
```